packages/protocol/scripts/bash/get_smart_contract_size_from_onchain_address.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+: '
+  Uses Celo CLI ("network:contracts") to get all core contracts and their implementation addresses.
+  Uses Foundry ("cast code") to get the bytecode deployed at each core contract implementation address.
+
+  Requirements:
+  1. Have Foundry installed
+  2. Have celocli installed
+  '
+
+# Get the current date and time for the filename
+TIMESTAMP=$(date +"%Y%m%d_%H%M%S")
+
+# Get the directory where the script is located, assuming the script is in the root or a subdirectory of the repo
+BASE_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+# Create the output directory if it doesn't exist
+OUTPUT_DIRECTORY="$BASE_DIRECTORY/out"
+mkdir -p "$OUTPUT_DIRECTORY"
+OUTPUT_FILE="$OUTPUT_DIRECTORY/onchain_bytecode_sizes_$TIMESTAMP.csv"
+TEMP_FILE="temp_$TIMESTAMP.txt"
+
+# Set the RPC URL for the blockchain
+RPC_URL="https://forno.celo.org"
```

Comment:
We might consider also supporting alfajores

Reply:
Sounds good, I added support for custom RPC URLs (e.g. `https://alfajores-forno.celo-testnet.org`) in https://github.com/celo-org/celo-monorepo/pull/10989/commits/7e5a6c6b8f52ceb8346aec75ea310408f5a3edac.

If no `RPC_URL` is given, it defaults to Mainnet.

I updated the `README` with instructions in https://github.com/celo-org/celo-monorepo/pull/10989/commits/1163dd97e709eef8eb24af012ab19a17122d0d79.